src/west/main.py

```diff
@@ -374,38 +373,44 @@ def set_zephyr_base(args):
         # At some point, we need a more flexible way to set environment
         # variables based on manifest contents, but this is good enough
         # to get started with and to ask for wider testing.
-        try:
-            manifest = Manifest.from_file()
-        except MalformedConfig as e:
-            log.die('Parsing of manifest file failed during command',
-                    args.command, ':', *e.args)
-        for project in manifest.projects:
-            if project.path == 'zephyr':
-                zb = project.abspath
-                zb_origin = 'manifest file {}'.format(manifest.path)
-                break
+        zb_env = os.environ.get('ZEPHYR_BASE')
+        zb_precedence = config.config.get('zephyr', 'zephyr-base-config',
```

Comment:
I think this should be:

```
[zephyr]
base-prefer=[env|config]
```

Reply:
I think we should do full wording on `configfile`,  cause an environment setting may also be a configuration somewhere.
I'm ok with `env` although I also think it is nice to have full word, as it is not something user changes all the time.

Reply:
I actually has been in doubt myself on the best word to choose.
So i'm ok with this as well.

Reply:
Or maybe just `autodetect=[env|configfile]`

Reply:
Then I like your previous suggestion better.
Cause not much _auto detection_ when we specify how to choose which value has precedence ;)

Reply:
update to use:
`env` and `base-prefer`.